R/prepare_survey_data.R

```diff
@@ -0,0 +1,196 @@
+        # Assuming interval censored people were circumcised before cens_year
+        circ_status = ifelse(yoc >= cens_year & circ_status == 1 & 
+                               !is.na(circ_age), 0.0, circ_status),
+        # circ censoring year / censor year in cens_year - 1 at cens_year - 1
+        yoc = ifelse(yoc == cens_year, cens_year - 1, yoc)
+      )
+  }
+  
+  ## Setting desired level aggregation ----------------------------------------
+  
+  # Getting the area level id to province
+  for (i in seq_len(max(areas$area_level))) {
+    survey_circumcision <- survey_circumcision %>%
+      # Merging on boundary information
+      left_join(
+        (areas %>%
+          st_drop_geometry() %>%
+          select(contains("area_id"), area_level)),
+        by = "area_id"
+      ) %>%
+      # Altering area
+      mutate(area_id = if_else(area_level == area_lev,
+                               as.character(area_id),
+                               as.character(parent_area_id))) %>%
+      select(-c(parent_area_id, area_level))
+  }
+  
+  ## Final preparation of circumcision variables ------------------------------
+  
+  # Preparing circumcision variables for the model
+  survey_circumcision <- survey_circumcision %>%
+    # Merging on the region index
+    left_join(
+      areas %>%
+        st_drop_geometry() %>%
+        select(area_id, area_name, space),
+      by = "area_id"
+    ) %>%
+    mutate(
+      # Time interval for the individual
+      time1 = yob - start_year + 1,
+      time2 = yoc - start_year + 1,
+      # Event type
+      event = ifelse(circ_status == 1 & !is.na(circ_age), 1,
+                     ifelse((circ_status == 1 & is.na(circ_age)), 2, 0)),
+      # Circumcision age
+      circ_age = yoc - yob,
+      age = circ_age + 1)
+  
+  # Adding circumcision type to dataset
+  survey_circumcision <- survey_circumcision %>%
+    # Type of circumcision
+    mutate(
+      circ_who = ifelse(circ_who == "other", 
+                        NA_character_, 
+                        circ_who),
+      circ_where = ifelse(circ_where == "other", 
+                          NA_character_, 
+                          circ_where),
+      type = case_when(
+        circ_who == "medical" | circ_where == "medical"         ~ "MMC",
+        circ_who == "traditional" | circ_where == "traditional" ~ "TMC",
+        TRUE                                                    ~ "Missing"
+      )
+    )
+  
+  # Getting surveys without any type information 
+  tmp <- with(survey_circumcision, as.data.frame(table(survey_id, type))) %>% 
+    group_by(survey_id) %>%
+    # calculate percentage and find surveys with all missing data
+    mutate(Freq = Freq / sum(Freq)) %>%
+    filter(type == "Missing", Freq == 1)
+  
+  # return warning detailing all surveys which are missing
+  if (nrow(tmp) > 1) {
+    for (i in seq_len(nrow(tmp))) {
+      warning(paste0(tmp$survey_id[i], " has all type == \"Missing\", and will 
```

Comment:
Up to you but Rich usually avoids warnings, see blog post here https://reside-ic.github.io/blog/a-warning-about-warning/

Reply:
Changed to a message, thanks for that. Very interesting blog, always thought the warning from `install.packages` for an unknown package name was ridiculous alright